x-pack/platform/plugins/shared/dataset_quality/server/routes/register_routes.ts

```diff
@@ -100,9 +98,27 @@ export function registerRoutes({
             opts.body.message = 'Client closed request';
           }
 
+          logRouteError(logger, opts);
+
           return response.customError(opts);
         }
       }
     );
   });
 }
+
+function logRouteError(
+  logger: Logger,
+  error: { statusCode: number; body: { message: string } } | Boom.Boom
+) {
+  const errorStatusCode = 'statusCode' in error ? error.statusCode : error.output.statusCode;
+  const errorMessage = Boom.isBoom(error) ? error.output.payload.message : error.body.message;
+
+  if (errorStatusCode >= 500) {
+    logger.error(errorMessage);
+  } else if (errorStatusCode < 500 && errorStatusCode >= 400) {
+    logger.warn(errorMessage);
```

Comment:
Hey @awahab07, based on the discussion in the meeting, shall we not log 4xx errors for now since it is not actionable for us?

Reply:
Would it be beneficial to have this logic more centralised in a shared plugin or it's better to define it within the plugins?

Reply:
> shall we not log 4xx errors for now

Sounds good, I'll adjust logging accordingly.

> Would it be beneficial to have this logic more centralised in a shared plugin or it's better to define it within the plugins? 

Absolutely, having it central would be ideal. In fact there's an issue https://github.com/elastic/kibana/issues/198681 which covers that.

Reply:
@maryam-saeidi any suggestions how 4xx should be logged then, logged as `debug` or not logged at all?

Reply:
For now, having it as debug sounds good to me until we work on https://github.com/elastic/kibana/issues/198681.